packages/db/src/lib/granule.ts

```diff
@@ -354,3 +360,61 @@ export const getGranulesByGranuleId = async (
     .where({ granule_id: granuleId });
   return records;
 };
+
+/**
+ * Change a granules' PG record and its files' PG record based on collection move
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction - DB client or transaction
+ * @param {Array<ApiGranuleRecord>} [granules] - updated ApiGranule records
+ * @returns {Promise<void>}
+ */
+export const updateGranulesAndFiles = async (
+  knexOrTransaction: Knex | Knex.Transaction,
+  granules: Array<ApiGranuleRecord>
+):Promise<void> => {
+  const {
+    granules: granulesTable,
+    files: filesTable,
+  } = TableNames;
+
+  try {
+    await createRejectableTransaction(knexOrTransaction, async (trx) => {
+      await Promise.all(granules.map(async (granule) => {
+        const pgGranule = await translateApiGranuleToPostgresGranule({
+          dynamoRecord: granule,
+          knexOrTransaction,
+        });
+        await trx(granulesTable).where('granule_id', '=', pgGranule.granule_id).update(
+          {
+            collection_cumulus_id: pgGranule.collection_cumulus_id,
+            updated_at: pgGranule.updated_at,
+            status: pgGranule.status,
+            last_update_date_time: pgGranule.last_update_date_time,
+          }
+        );
+        if (granule.files) {
```

Comment:
This doesn't look like it's handling file object set mismatches 'correctly'  - if we're going to expose this business logic as a DB package method, we need to make sure it handles that case correctly or put a logic-bound on making sure the behavior is understood.     

e.g. what does `updateGranulesAndFiles` behavior look like in the case where you have a files mismatch.   

Based on a first read through, it looks like this does a .update to only file records that match (e.g. already exist)... which means if you pass in a file object with *less* or more files, that's ignored.   

Documentation (or documentation via units) and guide-rails are probably needed here if we're writing a method that specific.